docs/slide-in-use-cases.md

```diff
@@ -0,0 +1,64 @@
+# Overview through Slide In use cases
+
+It turned out that the sliding-in feature is more complex than it had appeared in the first place.
+This paper is an overview through the use cases.
+It should help to implement and understand the feature.
+
+## Slide In New Branch Dialog
+
+The picture below shows the **Slide In New Branch Dialog**.
+This is a regular **Create New Branch Dialog** with a changed title.
+
+![Slide In New Branch Dialog](slide-in-new-branch-dialog.png "Slide In New Branch Dialog")
+
+Note the checkbox options:
+- Checkout branch
+- Overwrite existing branch
+
+## Use cases
+
+| UC ID | Branch entry exists | Local branch exists | Estimated frequency |
+| :---: | :---: | :---: | :---: |
+| 1 | false | false | ~ 75% |
+| 2 | false | true | ~ 25% |
+| 3 | true | true | < 1% |
+| 4 | true | false | < 1% |
+
+## Slide In - Logic
+
+- When the **entry does not exist**, it must be added (slid in) under the given branch.
+- When the **entry exists under the given parent**, branch layout must not be affected.
+- When the **entry exists under other parent**, it must be first removed (slid out),
```

Comment:
Also, an uncovered case here:

![image](https://user-images.githubusercontent.com/3383210/87126026-c8016a00-c28b-11ea-94c7-67524589a6eb.png)

**entry exists and is already root**, TBD what should be done (I think we could allow for attaching a root under a non-root branch, this might be useful e.g. for attaching `develop` under `master`)

Reply:
Also, I don't think the error message should say "...of **new** branch master" since it has already existed

Reply:
And also... I don't think more complex objects like `BranchLayout` should be passed to error messages like this one... I can address that on my follow-up PR